### ISSUE TITLE
v4.15.4: Unable to add locales in the admin panel

### ISSUE THREAD

**Issue by altrr2** [strapi/strapi#18822]
## Bug report

Unable to add locales in the admin panel

### Required System information

- Node.js version: 18.18.0 
- NPM version: 10.2.4
- Strapi version: 4.15.4
- Database: n/a
- Operating system: sonoma
- Is your project Javascript or Typescript: both affected

### Describe the bug

When adding locale in app.js (app.tsx) it does not appear on the admin login page

### Steps to reproduce the behavior

1. Clean install of the latest version 

```
npx create-strapi-app@latest my-project --quickstart
cd my-project
```

2. Add locale and run admin

```
cp ./src/admin/app.example.js ./src/admin/app.js
// uncomment one of the locales in the file, for example fr
npm run build && npm run develop
// only english appears in the top right corner of the admin page
```

3. downgrade strapi to v 4.15.0

```
npm i @strapi/plugin-i18n@4.15.0 @strapi/plugin-users-permissions@4.15.0 @strapi/strapi@4.15.0
npm run build && npm run develop
// now other enabled locale(s) are available
```

4. upgrade strapi back to v 4.15.4
```
npm i @strapi/plugin-i18n@4.15.4 @strapi/plugin-users-permissions@4.15.4 @strapi/strapi@4.15.4
npm run build && npm run develop
// only english
```

### Expected behavior

To use other locales when using >4.15.0

**Comment by derrickmehaffy** [comment]
confirmed on v4.15.4

**Comment by gfviegas** [comment]
I think this is duplicated from [#18786](https://github.com/strapi/strapi/issues/18786)

**Comment by poulato** [comment]
i get this error : Package path ./admin/src/selectors/selectI18nLocales is not exported from package! Please help

**Comment by joshuaellis** [comment]
resolved by #18764
